mslib/mscolab/server.py

```diff
@@ -71,6 +80,19 @@ class mscolab_auth:
                          ("add_new_user_here", "add_md5_digest_of_PASSWORD_here")]
         __file__ = None
 
+#setup idp login config
+if mscolab_settings.IDP_ENABLED :
+    with open("mslib/mscolab/app/mss_saml2_backend.yaml", encoding="utf-8") as fobj:
+        yaml_data = yaml.safe_load(fobj)
+    if os.path.exists("mslib/mscolab/app/idp.xml"):
+        yaml_data["config"]["sp_config"]["metadata"]["local"] = ["mslib/mscolab/app/idp.xml"]
+    else:
+        yaml_data["config"]["sp_config"]["metadata"]["local"] = []
+        warnings.warn("idp.xml file does not exists !")
```

Comment:
Is there any value in continuing to run if the file is missing? If not this could be simplified:

```suggestion
    if not os.path.exists(yaml_data["config"]["sp_config"]["metadata"]["local"][0]):
        sys.exit("idp.xml file does not exists !")
```

This would also remove the hardcoded path here and use the one specified in the configuration.

We should also consider if we want to merge the SP configuration file into the other mscolab configuration, but that can also be done later.

Reply:
@matrss In the fresh start when [configuring the IDP](https://github.com/nilupulmanodya/MSS/blob/Configure-mscolab-for-SSO/docs/conf_sso_test_msscolab.rst#configuring-the-mss-colab-server-and-identity-provider), it may be necessary to run the `/metadata` . However, if the idp.xml file is missing, the server will encounter an error. This is why I chose to take that approach. Do i need to change it ? If you have any additional suggestions, I would be grateful to hear them ? : )

Reply:
see my comments to the documentation.

Reply:
You are right, I forgot about the call to `/metadata` and we can not exit there. Still, we should not hardcode a path that is also set in the configuration. Something like this should be better:
```suggestion
    if not os.path.exists(yaml_data["config"]["sp_config"]["metadata"]["local"][0]):
        yaml_data["config"]["sp_config"]["metadata"]["local"] = []
        logging.warn("idp.xml file does not exists !")
```

Reply:
Further thought on that. 

I am not sure if we need the call of the server to fetch the metadata file and create it. It is may be also possible to use a template and fill in all data and return that rendered.  Of course we need to access the metadata by an url. But is it needed to have it saved to a file? When we save it we would need to recreate it on any cert change.